### PR TITLE
Fix contract account `sufficients` 

### LINKED ISSUE #140: Adjust sufficients for contract accounts
```rs
pub fn create_account(address: H160, code: Vec<u8>) {
    if code.is_empty() {
	return;
    }

    if !<AccountCodes<T>>::contains_key(address) {
	let account_id = T::AddressMapping::into_account_id(address);
	let _ = frame_system::Pallet::<T>::inc_sufficients(&account_id);
    }

    <AccountCodes<T>>::insert(address, code);
}
```
https://github.com/paritytech/frontier/blob/master/frame/evm/src/lib.rs#L694-L705

### PR DESCRIPTION
**Pull request by boundless-forest**
Fix #140 

I find there are some cases in which some evm accounts have code, but the sufficient is still 0 due to a previous frontier bug.
I hope we can fix it this time.

For example:

https://crab.subscan.io/account/0x0050f880c35c31c13bfd9cbb7d28aafaeca3abd2

image

### CODE REVIEW

**Comment by AurevoirXavier** [comment]
The `sufficients` inherit from the previous state. Why we need to increase this manually?

**Comment by boundless-forest** [comment]
> The `sufficients` inherit from the previous state. Why do we need to increase this manually?

Hmm, you are right. Close it.

**Comment by boundless-forest** [comment]
> The `sufficients` inherit from the previous state. Why do we need to increase this manually?

I have some new findings. Please check the description I added just now.

**Comment by boundless-forest** [comment]
https://github.com/paritytech/frontier/blob/master/frame/hotfix-sufficients/src/lib.rs

File: tool/state-processor/src/system/mod.rs [hunk 1]
```diff
@@ -111,6 +111,17 @@ impl Processor {
 			};
 
 			if let Some(k) = try_get_evm_address(&key) {
+				// If the evm account is a contract contract with sufficients, then we should
+				// increase the sufficients by one.
+				if self.solo_state.contains_key(&full_key(
+					b"EVM",
+					b"AccountCodes",
+					&blake2_128_concat_to_string(k),
```
Comment (AurevoirXavier) [inline tool/state-processor/src/system/mod.rs:119]: Cache this var to avoid double calculation.

**Pull request merged**